lib/matrix.gi

```diff
@@ -2153,6 +2153,22 @@ InstallMethod( NullspaceMat,
     [ IsOrdinaryMatrix ],
     mat -> SemiEchelonMatTransformation(mat).relations );
 
+InstallOtherMethod(NullspaceMat,"matrix objects",[IsMatrixObj],
+function(mat)
+local r,nr,nc,n,i,j;
+  r:=SemiEchelonMatTransformation(mat).relations;
+  if Length(r)=0 then return r;fi;
+  nr:=Length(r);
+  nc:=Length(r[1]);
+  n:=ZeroMatrix(BaseDomain(mat),nr,nc);
+  for i in [1..nr] do
+    for j in [1..nc] do
+      n[i,j]:=r[i][j];
+    od;
+  od;
+  return n;
+end);
```

Comment:
I think we have to define what `SemiEchelonMatTransformation` returns if the argument is a MatrixObj.
The above function assumes that `relations` contains a list of lists; is this reasonable?
(And in case that the nullspace is trivial, the result should be a MatrixObj.)

Reply:
Yes, this needs cleanup, and I felt I did not yet have sufficient experience with matrix objects to make this definition.